pkg/aws/aws.go

```diff
@@ -0,0 +1,169 @@
+	if !ok {
+		return nil, fmt.Errorf("s3 config for service %s not found", serviceName)
+	}
+	defaultOpts := []func(*s3.Options){
+		func(options *s3.Options) {
+			if s3Cfg.Region != "" {
+				options.Region = s3Cfg.Region
+			}
+			if credentialsSet(&s3Cfg.Credentials) {
+				options.Credentials = getCredentialsProvider(cfg, &s3Cfg.Credentials)
+			}
+			if s3Cfg.HttpClient.MaxIdleConns > 0 {
+				options.HTTPClient = createHttpClient(&s3Cfg.HttpClient)
+			}
+		},
+	}
+
+	opts = append(defaultOpts, opts...)
+
+	return s3.NewFromConfig(cfg, opts...), nil
+}
+
+func (b *Builder) NewSagemakerRuntimeService(
+	cfg aws.Config,
+	serviceName string, opts ...func(options *sagemakerruntime.Options)) (*sagemakerruntime.Client, error) {
+	sagemakerRuntimeCfg, ok := b.config.SagemakerRuntime[serviceName]
+	if !ok {
+		return nil, fmt.Errorf("sagemaker runtime config for service %s not found", serviceName)
+	}
+	defaultOpts := []func(*sagemakerruntime.Options){
+		func(options *sagemakerruntime.Options) {
+			if sagemakerRuntimeCfg.Region != "" {
+				options.Region = sagemakerRuntimeCfg.Region
+			}
+			if credentialsSet(&sagemakerRuntimeCfg.Credentials) {
+				options.Credentials = getCredentialsProvider(cfg, &sagemakerRuntimeCfg.Credentials)
+			}
+			if sagemakerRuntimeCfg.HttpClient.MaxIdleConns > 0 {
+				options.HTTPClient = createHttpClient(&sagemakerRuntimeCfg.HttpClient)
+			}
+		},
+	}
+
+	opts = append(defaultOpts, opts...)
+
+	return sagemakerruntime.NewFromConfig(cfg, opts...), nil
+}
+
+func (b *Builder) NewSfnService(
+	cfg aws.Config,
+	serviceName string, opts ...func(options *sagemakerruntime.Options)) (*sagemakerruntime.Client, error) {
+	sfnCfg, ok := b.config.Sfn[serviceName]
+	if !ok {
+		return nil, fmt.Errorf("sfn config for service %s not found", serviceName)
+	}
+	defaultOpts := []func(*sagemakerruntime.Options){
+		func(options *sagemakerruntime.Options) {
+			if sfnCfg.Region != "" {
+				options.Region = sfnCfg.Region
+			}
+			if credentialsSet(&sfnCfg.Credentials) {
+				options.Credentials = getCredentialsProvider(cfg, &sfnCfg.Credentials)
+			}
+			if sfnCfg.HttpClient.MaxIdleConns > 0 {
+				options.HTTPClient = createHttpClient(&sfnCfg.HttpClient)
+			}
+		},
+	}
+
+	opts = append(defaultOpts, opts...)
+
+	return sagemakerruntime.NewFromConfig(cfg, opts...), nil
+}
+
+func (b *Builder) NewSqsService(
+	cfg aws.Config, serviceName string, opts ...func(options *s3.Options)) (*s3.Client, error) {
+	sqsCfg, ok := b.config.Sqs[serviceName]
+	if !ok {
+		return nil, fmt.Errorf("sqs config for service %s not found", serviceName)
+	}
+	defaultOpts := []func(*s3.Options){
+		func(options *s3.Options) {
+			if sqsCfg.Region != "" {
+				options.Region = sqsCfg.Region
+			}
+			if credentialsSet(&sqsCfg.Credentials) {
+				options.Credentials = getCredentialsProvider(cfg, &sqsCfg.Credentials)
+			}
+			if sqsCfg.HttpClient.MaxIdleConns > 0 {
+				options.HTTPClient = createHttpClient(&sqsCfg.HttpClient)
+			}
+		},
+	}
```

Comment:
@laynax 

Thanks for the proposal. But having this would require us to have `with*` functions per service, like `withS3Region`, `withS3Credentials` etc. Since it is a private functionality and not exposed, I suggest keep as is